src/overlays/actors/ovl_En_Sda/z_en_sda.c

```diff
@@ -28,18 +32,327 @@ const ActorInit En_Sda_InitVars = {
+                *shadowTextureTemp = 255;
+            } else {
+                *shadowTextureTemp = 0;
+            }
+        }
+    } else {
+        for (shadowTextureTemp32 = (s32*)shadowTexture, i = 0; i < 0x400; i++, shadowTextureTemp32++) {
+            *shadowTextureTemp32 = 0;
+        }
+    }
+
+    Matrix_SetStateXRotation((BREG(50) + 70) / 100.0f);
+
+    for (i = 0; i < 18; i++) {
+        if (D_80947AFC[i] >= 0) {
+            D_80947EA0[D_80947AFC[i]] = player->bodyPartsPos[i];
+        }
+    }
+
+    D_80947EA0[0].y += 3.0f;
+    D_80947EA0[15].x = D_80947EA0[0].x + ((D_80947EA0[15].x - D_80947EA0[0].x) * 1.2f);
+    D_80947EA0[15].y = D_80947EA0[0].y + ((D_80947EA0[15].y - D_80947EA0[0].y) * -1.2f);
+    D_80947EA0[15].z = D_80947EA0[0].z + ((D_80947EA0[15].z - D_80947EA0[0].z) * 1.2f);
+
+    for (i = 0; i < 6; i++) {
+        func_809469C0(player, shadowTexture, i / 5.0f);
+    }
+
+    if (ENSDA_GET(&this->actor) != ENSDA_1) {
+        func_8018219C(&player->shieldMf, &sp178, false);
+
+        sp178.y += (KREG(87) * 0x8000) + 0x8000;
+        sp178.x *= (KREG(88) - 1);
+
+        Matrix_InsertMatrix(&player->shieldMf, MTXMODE_NEW);
+        Matrix_MultiplyVector3fByState(&D_80947A60, &sp16C);
+        Matrix_RotateY(sp178.y, MTXMODE_NEW);
+        Matrix_InsertXRotation_s(sp178.x, MTXMODE_APPLY);
+
+        for (i = 0; i < 22; i++) {
+            Matrix_MultiplyVector3fByState(&D_80947B10[i], &sp188);
+            sp64[i].x = (((KREG(82) / 100.0f) + 4.0f) * sp188.x) + sp16C.x;
+            sp64[i].y = (((KREG(82) / 100.0f) + 4.0f) * sp188.y) + sp16C.y;
+            sp64[i].z = (((KREG(82) / 100.0f) + 4.0f) * sp188.z) + sp16C.z;
+        }
+
+        Matrix_SetStateXRotation((BREG(50) + 70) / 100.0f);
+
+        for (i = 0; i < 22; i++) {
+            sp194.x = sp64[i].x - player->actor.world.pos.x;
+            sp194.y = sp64[i].y - player->actor.world.pos.y + KREG(80) + 16.0f;
+            sp194.z = sp64[i].z - player->actor.world.pos.z;
+
+            Matrix_MultiplyVector3fByState(&sp194, &sp188);
+
+            sp188.x *= (1.0f + (KREG(90) / 100.0f));
+            sp188.y *= (1.0f + (KREG(90) / 100.0f));
+            temp_t0 = sp188.x + 32.0f;
+            temp_t1 = sp188.y * 64.0f;
+
+            do {
+                for (j = 0, phi_a3 = -0xC0; j < 7; j++, phi_a3 += 0x40) {
+                    for (phi_a0 = -D_80947A78[j]; phi_a0 < D_80947A78[j] - 1; phi_a0++) {
+                        temp_v0 = temp_t0 + phi_a0;
+                        if ((temp_v0 >= 0) && (temp_v0 < 0x40)) {
+                            temp_v1 = temp_t1 + phi_a3;
+                            if ((temp_v1 >= 0) && (temp_v1 < 0x1000)) {
+                                shadowTexture[temp_v0 + temp_v1] = 255;
+                            }
+                        }
+                    }
+                }
+                j++;
+            } while (j < 6);
```

Comment:
What is this.... j gets incremented beyond the loop? Where does the loop begin and where does it end? 
So weird.